collections/zip.ts

```diff
@@ -31,7 +31,7 @@ import { minOf } from "./min_of.ts";
 export function zip<T extends unknown[]>(
   ...arrays: { [K in keyof T]: T[K][] }
 ): T[] {
-  const minLength = minOf(arrays, (it) => it.length) ?? 0;
+  const minLength = minOf(arrays, (it) => it.length)!;
```

Comment:
This value can be undefined when the input is an empty array. So this non-null assertion doesn't look good to me. If we want to increase the branch coverage here, we can call `zip()` and that should cover `?? 0` part.

Reply:
Ah, my previous assumption was that this wasn't the case. Thanks!